lib/utils/utf8.c

```diff
@@ -79,3 +81,32 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+bool utf8_check(const char *str, size_t maxlen)
+{
+	int i = 0, nbyte = 0;
+	int len = strnlen(str, maxlen);
+	unsigned char *buf = (unsigned char *)str;
```

Comment:
This should be const as we are not modifying the string.

Reply:
It will trigger a ci warning, that is STATIC_CONST.

Reply:
Usually you should never cast away const pointer as that would indicate that you could modify the thing the pointer points to. Usually compiler should warn about this, and I recall some static checkers will.
So it is hard to understand why
```
const unsigned char *buf = (const unsigned char *)str;
```
would give a warning.

Reply:
Unresolving this. https://github.com/zephyrproject-rtos/zephyr/pull/80779#discussion_r1828544458 claims it's answered here, but I disagree. Accessing `str` directly should be possible without the additional local `buf` and should be possible to use `const`. 

Please provide additional information about the warning you see

Reply:
I won't add const, jukkar has already tested it. This will make my PR fail CI.

Reply:
@rruuaanng please read https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#workflow-suggestions-that-help-reviewers - Don't resolve comments that reviewers disagree with. 

You have not provided any reason except "it fails CI" - Please provide the evidence on why this is an issue. It really should not be a problem to use `str` directly in this function. 

You may be hitting a CI issue I've seen before where `const char myval = "some value"` isn't allowed, because it's not `static`. If that is what you are seeing, we should create a GH issue and use a workaround in the unit tests (which I ended up doing), instead of the implementation.

Reply:
CI requires it to be a static variable(STATIC_CONST), but in fact it will not compile if declared as static.

Reply:
![Uploading 图片.png…]()

Reply:
@rruuaanng I am once again going to ask you _not_ to resolve comments. 

In the implementation you don't need the `buf` at all, you can just use `str` directly. Are you seeing issues with that approach?

Reply:
```
printf("%d %d %d", 0xe4, (const char)'\xe4', (const unsigned char)'\xe4');

output:
228 -28 228
```
I need to reiterate that you should read the above review carefully :)

Reply:
Can you share the log of the build warning/error you get? https://github.com/zephyrproject-rtos/zephyr/pull/80779#discussion_r1829145564 is just a screenshot of your editor, which doesn't really say much :) 

The other utf8 functions works fine with `const char *`, so I'm curious to see what the issue here is

Reply:
In fact, other implementations have problems. They do not check the extended bytes or the abnormal bytes. The above case has explained why I want to convert it to unsigned.

Reply:
> I won't add const, jukkar has already tested it. This will make my PR fail CI.

I have not said anything about testing or accepting this, please don't claim that.

What I have said is that you should not cast away constness of a variable. I don't get why you talk about static variable in this case.